packages/main/src/plugin/container-registry.ts

```diff
@@ -170,20 +172,11 @@ export class ContainerProviderRegistry {
       name: provider.name,
       connection: containerProviderConnection,
     };
-    let previousStatus = containerProviderConnection.status();
 
-    if (containerProviderConnection.status() === 'started') {
-      internalProvider.api = new Dockerode({ socketPath: containerProviderConnection.endpoint.socketPath });
-      if (containerProviderConnection.type === 'podman') {
-        internalProvider.libpodApi = internalProvider.api as unknown as LibPod;
-      }
-      this.handleEvents(internalProvider.api);
-    }
-
-    // track the status of the provider
-    const timer = setInterval(async () => {
-      const newStatus = containerProviderConnection.status();
-      if (newStatus !== previousStatus) {
+    providerRegistry.onBeforeDidUpdateContainerConnection(event => {
+      if (event.providerId === provider.id && event.connection.name === containerProviderConnection.name) {
+        const newStatus = event.status;
+        console.log(` New status ${newStatus}`);
```

Comment:
```suggestion
        console.log(` New status ${newStatus}`);
```

not sure to keep it ?